ddtrace/contrib/requests/connection.py

```diff
@@ -17,23 +18,16 @@
 
 def _extract_hostname(uri):
     # type: (str) -> str
-    end = len(uri)
-    j = uri.rfind("#", 0, end)
-    if j != -1:
-        end = j
-    j = uri.rfind("&", 0, end)
-    if j != -1:
-        end = j
-
-    start = uri.find("://", 0, end) + 3
-    i = uri.find("@", start, end) + 1
-    if i != 0:
-        start = i
-    j = uri.find("/", start, end)
-    if j != -1:
-        end = j
-
-    return uri[start:end]
+    parsed_uri = parse.urlparse(uri)
+    port = None
+    try:
+        port = parsed_uri.port
+    except ValueError:
+        pass
```

Comment:
```suggestion
        return "invalid_url"
```

Reply:
I would suggest we use `?` in something like the service as it will be in the UI where a longer string is less desirable. So in the case that a requests get is to a url with an invalid port, we'd have a request span marked as an error and the service, if split by domain is enabled, would be `hostname:?` rather than `hostname:8888` otherwise.